### PR TITLE
feat: create user information form in checkout page

### PR DESCRIPTION
**Pull request by TonNQ**
Show seats selected and create form for filling user information in

86epq424h

### CODE REVIEW

File: public/mockData/movie_detail.json
Comment (ces-anhlam) [inline public/mockData/movie_detail.json:1]: This data is very different from the actual data that will be returned. Please adjust this data to match the DTO.

File: public/mockData/showtimes.json
Comment (ces-anhlam) [inline public/mockData/showtimes.json:1]: same above

File: src/apis/fetch.api.js [hunk 1]
```diff
@@ -1,10 +1,15 @@
-import { toast } from 'react-toastify'
-
-export const fetchApi = async (fileName) => {
-  try {
-    const response = await fetch(`/mockData/${fileName}.json`)
-    return response.json()
-  } catch (error) {
-    toast.error(error)
-  }
+export function fetchApi(fileName) {
```
Comment (ces-anhlam) [inline src/apis/fetch.api.js:1]: use async await

File: src/components/Input/Input.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export default function Input({
```
Comment (ces-anhlam) [inline src/components/Input/Input.jsx:1]: TextInput ?

File: src/components/UserInfoForm/UserInfoForm.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    handleSubmit,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema)
+  })
+  const dispatch = useDispatch()
+  const hiddenSubmitButtonRef = useRef(null)
+
+  const onSubmit = handleSubmit((data) => {
+    console.log(data)
```
Comment (ces-anhlam) [inline src/components/UserInfoForm/UserInfoForm.jsx:25]: remove log and add TODO if you don't have API now

File: src/utils/datetime.js [hunk 4]
```diff
@@ -43,3 +36,32 @@ export const getHourAndMinute = (time) => {
...
+    return null
+  }
+}
+
+export const getMinuteAndSecond = (remainingSeconds) => {
+  const duration = moment.duration(remainingSeconds, 'seconds')
+  const minutes = Math.floor(duration.asMinutes())
+  const seconds = duration.seconds()
+
+  const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes
```
Comment (ces-anhlam) [inline src/utils/datetime.js:53]: use moment format this

File: src/apis/fetch.api.js (hunk 1 above)
Comment (TonNQ) [reply src/apis/fetch.api.js:1]: fixed

File: src/components/Input/Input.jsx (hunk 2 above)
Comment (TonNQ) [reply src/components/Input/Input.jsx:1]: Yes, it's for only text input. I changed this component's name.

File: public/mockData/movie_detail.json
Comment (TonNQ) [reply public/mockData/movie_detail.json:1]: fixed

File: public/mockData/showtimes.json
Comment (TonNQ) [reply public/mockData/showtimes.json:1]: fixed

File: src/components/UserInfoForm/UserInfoForm.jsx (hunk 3 above)
Comment (TonNQ) [reply src/components/UserInfoForm/UserInfoForm.jsx]: fixed

File: src/utils/datetime.js (hunk 4 above)
Comment (TonNQ) [reply src/utils/datetime.js]: fixed

**Review by ces-khanhhoang — APPROVED** [review]

File: src/components/UserInfoForm/UserInfoForm.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    control,
+    handleSubmit,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema)
+  })
+  const dispatch = useDispatch()
+  const hiddenSubmitButtonRef = useRef(null)
+
+  const onSubmit = handleSubmit((data) => {
```
Comment (ces-anhlam) [inline src/components/UserInfoForm/UserInfoForm.jsx:24]: fix as we discuss

File: src/components/UserInfoForm/UserInfoForm.jsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        errorMessage={errors.email?.message}
+      />
+      <div className='mb-2'>
+        <label className='flex items-center gap-2 hover:cursor-pointer'>
+          <Controller
+            name='verifyAge'
+            control={control}
+            defaultValue={false}
+            render={({ field }) => (
+              <input type='checkbox' {...field} className='h-[16px] w-[16px]' checked={field.value} />
```
Comment (ces-anhlam) [inline src/components/UserInfoForm/UserInfoForm.jsx:78]: create checkbox component

File: src/utils/price.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+export const formatPrice = (price, decimalSeparator = '.', thousandSeparator = ',') => {
```
Comment (ces-anhlam) [inline src/utils/price.js:1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat
Can you try this

**Review by ces-phatnguyen — APPROVED** [review]

**Pull request merged**